agent.go

```diff
@@ -212,3 +264,33 @@ func getBoolEnv(key string, fallback bool) bool {
 	}
 	return value
 }
+
+func getGitData() *GitData {
+	gitDataOnce.Do(func() {
+		gitData = GetCurrentGitData()
```

Comment:
Does this actually work? I would think this would only be executed once and `gitData` would be undefined the second time it runs

Reply:
Yes, It works because I'm storing the first run result to the `gitData` var, then in other runs it will use the data calculated the first time.

Reply:
Ok, I was confused because of how github has folded the source code - my bad